Makefile

```diff
@@ -12,7 +12,7 @@ SOEXT ?= $(shell ruby -e 'puts RbConfig::CONFIG["SOEXT"]')
 
 CPPFLAGS := -Iinclude $(CPPFLAGS)
 CFLAGS := -g -O2 -std=c99 -Wall -Werror -Wextra -Wpedantic -Wundef -Wconversion -Wno-missing-braces -fPIC -fvisibility=hidden $(CFLAGS)
-CC ?= cc
+ARFLAGS := -r
```

Comment:
I introduced `?=` in #2705 to be explicit that the variable may be overwritten by env vars sent from `extconf.rb`.
But a quick test shows that `<implicit variable> ?= <anything>` doesn’t do anything because the variable was already set – implicitly by GNU `make`! :sweat_smile:
This is why this assignment must be either `=` or `:=` but not `?=`. It shouldn’t affect receiving the `rbconfig` `ARFLAGS` through the aforementioned env vars.

Reply:
* [ ] TODO: circument __only__ for `ar -rv`; do not change for `libtool ` (such as used in that *one* failing CI)

Reply:
Switched to `--no-builtin-variables`. Explicit is better than implicit anyways.